src/ai/backend/manager/models/image.py

```diff
@@ -137,17 +137,30 @@ def _apply_loading_option(
     return query_stmt
 
 
-async def load_all_registries(
+async def load_configured_registries(
     db: ExtendedAsyncSAEngine,
+    project: Optional[str],
 ) -> dict[str, ContainerRegistryRow]:
     join = functools.partial(join_non_empty, sep="/")
 
     async with db.begin_readonly_session() as session:
         result = await session.execute(sa.select(ContainerRegistryRow))
-        all_registry_config = {
-            join(row.registry_name, row.project): row for row in result.scalars().all()
-        }
-    return cast(dict[str, ContainerRegistryRow], all_registry_config)
+        if project:
+            registries = cast(
+                dict[str, ContainerRegistryRow],
+                {
+                    join(row.registry_name, row.project): row
+                    for row in result.scalars().all()
+                    if row.project == project
+                },
+            )
+        else:
+            registries = cast(
+                dict[str, ContainerRegistryRow],
+                {join(row.registry_name, row.project): row for row in result.scalars().all()},
+            )
```

Comment:
Creating a filter and passing it could generalize the approach, but let’s leave it as is for now.